scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,219 @@
 package scenelib.annotations.el;
 
+import org.objectweb.asm.Label;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
 import java.util.Objects;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 /**
  * A {@link LocalLocation} holds location information for a local
- * variable: slot index, scope start, and scope length.
+ * variable: slot index, scope start, and scope end.
  */
 public final class LocalLocation {
     /**
-     * The slot index of the local variable.
+     * The starts of the scopes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
+     */
+    public final Label[] start;
+
+    /**
+     * The ends of the scopes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int index;
+    public final Label[] end;
 
     /**
-     * The start of the local variable's scope (or live range), as an offset
-     * from the beginning of the method code in bytes.
+     * The indices of the element being visited in the classfile.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeStart;
+    public final int[] index;
 
     /**
-     * The length of the local variable's scope (or live range), in bytes.
+     * The name of the local variable being visited.
      */
-    public final int scopeLength;
+    public final @Nullable String variableName;
 
-    public final String varName;
-    public final int varIndex;
+    // We now use the public ASM bytecode framework instead of our own local copy.
+    // As part of the implementation changes, the following member variables
+    // of LocalLocation no longer exist:
+    //   int scopeStart;
+    //   int scopeLength;
+    //   int varIndex;
+    // Instead, the following accessor methods must be used:
+    //   int getScopeStart();
+    //   int getScopeLength();
+    //   int getVarIndex();
 
     /**
-     * Constructs a new {@link LocalLocation}; the arguments are assigned to
-     * the fields of the same names.
+     * Construct a new LocalLocation.
+     *
+     * @param variableName the name of the local variable
+     * @param index the offset of the variable in the stack frame
      */
-    public LocalLocation(int index, int scopeStart, int scopeLength) {
+    public LocalLocation(String variableName, int index) {
+        this(new Label[] {new Label()}, new Label[] {new Label()}, new int[] {index}, variableName);
+    }
+
+    /**
+     * Construct a new LocalLocation.
+     *
+     * @param start thes start of the variable's lifetime(s)
+     * @param end the ends of the variable's lifetime(s)
+     * @param index the stack offsets of the variable's lifetime(s)
```

Comment:
Are these offsets or indices?  (And is there a difference between the two?)

Reply:
offsets - added to javadoc info